improver_tests/cli/test_init.py

```diff
@@ -197,111 +196,289 @@ def test_with_output(self, m):
+        find a full match.
+        Calls load_cube three times due to the first match not being
+        the same length as the input_list.
+
+        The returned result is the last two arguments used by the load_cube
+        Mocked side_effect.
+        """
+        constrained_list = create_constrained_inputcubelist_converter(
+            ['cats'],
+            ['cats', self.direction_name])
+        result = constrained_list(self.fake_path)
+        raw.assert_called_once_with(self.fake_path)
+        for constr in ['cats', 'cats', self.direction_name]:
+            load.assert_any_call(self.fake_path, constraints=constr)
+        # Calls 3 times, as the first loads but is wrong size, second matches
+        self.assertEqual(load.call_count, 3)
+        self.assertEqual(len(result), 2)
+        self.assertEqual(result[0], 'cube2')
+        self.assertEqual(result[1], 'cube3')
+
+    @patch('iris.load_raw', side_effect=[cubelist_of_length(2)])
+    @patch('improver.utilities.load.load_cube', side_effect=[ValueError,
+                                                             ValueError])
+    def test_err_when_no_match(self, load, raw):
+        """Tests that raises an error when no cubes match any constraints.
+        Tests that load_cube is called twice.
+        The 'filepath' and constraints as arguments.
+        A match isn't found so a ValueError is returned.
+        """
+        constrained_list = create_constrained_inputcubelist_converter(
+            [self.speed_name], [self.direction_name])
+        msg = 'A full list'
         with self.assertRaisesRegex(ValueError, msg):
-            obj(self.wind_cubes)
-
-    @patch('improver.cli.maybe_coerce_with', new=replace_load_with_extract)
-    def test_list_no_match(self):
-        """Tests that providing no valid constraints raises a ValueError."""
-        obj = create_constrained_inputcubelist_converter(['nonsense'])
-        msg = 'Incorrect number of valid inputs available'
+            constrained_list(self.fake_path)
+        raw.assert_called_once_with(self.fake_path)
+        for constr in [self.speed_name, self.direction_name]:
+            load.assert_any_call(self.fake_path, constraints=constr)
+        # Calls twice more as each times it makes a call that returns ValueErr
+        self.assertEqual(load.call_count, 2)
+
+    @patch('iris.load_raw', side_effect=[cubelist_of_length(2)])
+    @patch('improver.utilities.load.load_cube',
+           side_effect=cubelist_of_length(1))
+    def test_err_when_match_wrong_size(self, load, raw):
+        """Tests that raises an error when there isn't a full match of
+        constraint to cube.
+        The cubelist has two cubes but the constraint list only has 1 name.
+        Therefore not all cubes were able to be extracted.
+        """
+        constrained_list = create_constrained_inputcubelist_converter(
+            [self.speed_name])
+        # . needed to replace square brackets[]
```

Comment:
?

Reply:
The regex used to be different before i remembered Python needed a double \\[.
```suggestion
```